### ISSUE TITLE
Regression 0.143.0 for JSONPath numeric filter with string values

### ISSUE THREAD

**Issue by cburgmer** [danielaparker/jsoncons#203]
There seems to be a regression in 0.1.43.0 for `$[?(@.key<42)]` if a string value is involved
(compare https://cburgmer.github.io/json-path-comparison/):

- [ ] `$[?(@.key<42)]`
  Input:
  ```
  [{"key": 0}, {"key": 42}, {"key": -1}, {"key": 41}, {"key": 43}, {"key": 42.0001}, {"key": 41.9999}, {"key": 100}, {"some": "value"}]
  ```
  Expected output:
  ```
  [{"key": 0}, {"key": -1}, {"key": 41}, {"key": 41.9999}]
  ```
  Actual output:
  ```
  [{"key": 0}, {"key": -1}, {"key": 41}, {"key": 41.9999}, {"some": "value"}]
  ```

As there is no definite standard on JSONPath, those edge cases require interpretation by implementations. The "expected output" is however recorded by the majority of implementations.

For reference, the output was generated by the program in https://github.com/cburgmer/json-path-comparison/tree/master/implementations/Cpp_jsoncons.

**Comment by cburgmer** [comment]
You might be interested to know that the comparison project aims to be usable as a regression test suite: https://github.com/cburgmer/json-path-comparison#regression-test-suite.

**Comment by danielaparker** [comment]
Thanks! It looks like jsoncons unit  test coverage isn't comprehensive enough. I've deleted 0.1.43.0 until this is sorted out.

**Comment by danielaparker** [comment]
Fixed in v0.143.1.
